test/groovy/MTABuildTest.groovy

```diff
@@ -16,31 +19,40 @@ import util.Rules
 
 public class MTABuildTest extends BasePipelineTest {
 
+    @ClassRule
+    public static TemporaryFolder tmp = new TemporaryFolder()
+
     private ExpectedException thrown = new ExpectedException()
-    private TemporaryFolder tmp = new TemporaryFolder()
     private JenkinsLoggingRule jlr = new JenkinsLoggingRule(this)
     private JenkinsShellCallRule jscr = new JenkinsShellCallRule(this)
 
     @Rule
     public RuleChain ruleChain = Rules.getCommonRules(this)
             .around(thrown)
-            .around(tmp)
             .around(jlr)
             .around(jscr)
 
-
-    def currentDir
-    def mtaYaml
+    private static currentDir
+    private static newDir
+    private static mtaYaml
 
     def mtaBuildScript
     def cpe
 
+    @BeforeClass
+    static void createTestFiles() {
+
+        currentDir = "${tmp.getRoot()}"
+        mtaYaml = tmp.newFile('mta.yaml')
+        newDir = "$currentDir/newDir"
+        tmp.newFolder('newDir')
+        tmp.newFile('newDir/mta.yaml') << defaultMtaYaml()
+    }
+
     @Before
     void init() {
 
-        currentDir = tmp.newFolder().toURI().getPath()[0..-2] //omit final '/'
-        mtaYaml = new File("$currentDir/mta.yaml")
-        mtaYaml << defaultMtaYaml()
+        mtaYaml.text = defaultMtaYaml()
```

Comment:
[nit] would also work with the "<<" way as it was up to now

Reply:
No, this does not work, that's why I did change it.